VideoLocker/src/main/java/org/edx/mobile/view/adapters/DiscussionPostsAdapter.java

```diff
@@ -46,60 +38,52 @@ public void render(BaseViewHolder tag, DiscussionThread discussionThread) {
             }
-            holder.discussionPostPinFollowTextView.setText(pinFollowTextLabel);
-
-            // If the discussion thread has an extra line, add 25dp to row height
-            // TODO: Add the additional height if the title spans 2 lines
-            AbsListView.LayoutParams params = (AbsListView.LayoutParams) holder.discussionPostRow.getLayoutParams();
-            int boxHeight = context.getResources().getDimensionPixelSize(
-                    R.dimen.discussion_post_box_height);
-            int extraHeight = context.getResources().getDimensionPixelSize(
-                    R.dimen.discussion_post_extra_box_height);
-            params.height = boxHeight + extraHeight;
-            holder.discussionPostRow.setLayoutParams(params);
+            final Drawable icon = discussionThread.isPinned() ?
+                    new IconDrawable(getContext(), Iconify.IconValue.fa_thumb_tack).colorRes(R.color.edx_grayscale_neutral_base).sizeRes(R.dimen.edx_x_small)
+                    : null;
+            holder.discussionPostPinTextView.setCompoundDrawablesWithIntrinsicBounds(icon, null, null, null);
+            holder.discussionPostPinTextView.setText(pinFollowTextLabel);
+
+        } else {
+            holder.discussionPostPinTextView.setVisibility(View.GONE);
+        }
+
+        if (discussionThread.isFollowing()) {
+            holder.discussionPostFollowTextView.setVisibility(View.VISIBLE);
+            holder.discussionPostFollowTextView.setCompoundDrawablesWithIntrinsicBounds(
+                    new IconDrawable(getContext(), Iconify.IconValue.fa_star).colorRes(R.color.edx_grayscale_neutral_base).sizeRes(R.dimen.edx_x_small), null, null, null);
+        } else {
+            holder.discussionPostFollowTextView.setVisibility(View.GONE);
         }
 
-        int commentColor = discussionThread.getUnreadCommentCount() == 0 ?
+        final int commentColor = discussionThread.getUnreadCommentCount() == 0 ?
                 R.color.edx_grayscale_neutral_light : R.color.edx_brand_primary_base;
         holder.discussionPostNumCommentsTextView.setText(Integer.toString(discussionThread.getCommentCount()));
-        holder.discussionPostNumCommentsTextView.setTextColor(context.getResources().getColor(commentColor));
+        holder.discussionPostNumCommentsTextView.setTextColor(getContext().getResources().getColor(commentColor));
         holder.discussionPostCommentIcon.setIcon(Iconify.IconValue.fa_comment);
-        holder.discussionPostCommentIcon.setIconColor(context.getResources().getColor(commentColor));
-
-        // TODO: Read property is not being returned in API to get the thread list
-        if (discussionThread.isRead()) {
-            holder.discussionPostRow.setBackgroundColor(context.getResources().getColor(
-                    R.color.edx_grayscale_neutral_xx_light));
-        }
+        holder.discussionPostCommentIcon.setIconColor(getContext().getResources().getColor(commentColor));
```

Comment:
The color changes depending on the # of unread comments so must be set dynamically.

I moved the icon setting to XML though